zerver/lib/thumbnail.py

```diff
@@ -26,6 +26,18 @@ def user_uploads_or_external(url: str) -> bool:
     )
 
 
+def image_url_is_trusted(url: str, realm_host: str) -> bool:
```

Comment:
How about `is_static_or_current_realm_url` for this function?  And I think `subdomains.py` -- where we have other security model functions related to URLs -- would be more appropriate, since this function doesn't have to do with thumbnailing or images, even if the place we use it is related to those things.

I'd also accept the `Realm` object as the second parameter, rather than the string; I think that's a cleaner calling convention.

One you've addressed this, I'd like a final security review from @andersk.